### ISSUE TITLE
tour: Validate method doesn't validate number of bytes written

### ISSUE THREAD

**Issue by CoolOppo** [golang/tour#643]
Context: https://tour.golang.org/methods/22

```go
package main

import "golang.org/x/tour/reader"

type MyReader struct{}

func (MyReader) Read(b []byte) (n int, err error) {
	n, err = 0, nil
	for i := range b {
		b[i] = byte('A')
		n += 1
	}
	return 1, nil // SHOULD NOT SUCCESSFULLY VALIDATE
}

func main() {
	reader.Validate(MyReader{})
}
```

Output:  
```
OK!
```

In this example, `Read` will always return that it wrote to 1 element, and as such, shouldn't pass the test.

**Comment by crisman** [comment]
It does and should pass the test.  Since you do put `'A'` in `b[0]` each time and then report 1 byte read everything is fine.  The only issue is that the for loop inside `Read()` is wasting time putting values that nothing will look at in `b[1]` and above.

It would be an error if `Read()` was reading from a known text, but since all we are doing is ignoring some part of the infinite `'A'`, we still get an infinite set of them.

**Comment by ALTree** [comment]
I agree this implementation does not violate `Read`'s contract, closing here.
